src/Microsoft.AspNet.Mvc.Razor/RazorViewEngineOptionsSetup.cs

```diff
@@ -17,16 +18,26 @@ public class RazorViewEngineOptionsSetup : ConfigureOptions<RazorViewEngineOptio
         /// Initializes a new instance of <see cref="RazorViewEngineOptions"/>.
         /// </summary>
         /// <param name="applicationEnvironment"><see cref="IApplicationEnvironment"/> for the application.</param>
-        public RazorViewEngineOptionsSetup(IApplicationEnvironment applicationEnvironment)
-            : base(options => ConfigureRazor(options, applicationEnvironment))
+        /// <param name="hostingEnvironment"><see cref="IHostingEnvironment"/> for the application.</param>
+        public RazorViewEngineOptionsSetup(IApplicationEnvironment applicationEnvironment, IHostingEnvironment hostingEnvironment = null)
+            : base(options => ConfigureRazor(options, applicationEnvironment, hostingEnvironment))
         {
         }
 
         private static void ConfigureRazor(
             RazorViewEngineOptions razorOptions,
-            IApplicationEnvironment applicationEnvironment)
+            IApplicationEnvironment applicationEnvironment,
+            IHostingEnvironment hostingEnvironment)
         {
             razorOptions.FileProvider = new PhysicalFileProvider(applicationEnvironment.ApplicationBasePath);
+            if (hostingEnvironment != null && hostingEnvironment.IsDevelopment())
+            {
+                razorOptions.Configuration = "Release";
```

Comment:
Looks like the values are inverted for the case (`IsDevelopment` -> `Debug` else `Release`).